packages/types/index.d.ts

```diff
@@ -0,0 +1,57 @@
+// index.d.ts
+
+export type Input =
+  | {
+      prompt: string;
+    }
+  | {
+      messages: ChatMessage[];
+    };
+
+export type Output =
+  | {
+      text: string;
+    }
+  | {
+      message: ChatMessage;
+    };
+
+type LLM =
```

Comment:
is there a way to do this with an enum (even though it's technically an object type)? wondering if we should make it easier for people to enumerate over the ModelIDs, which you can't do this way.

The best types are in `extension/src/public-interface.ts`

Reply:
yeah I think we can do this with our own lib or rely on a community contribution